tools/hsmtool.c

```diff
@@ -368,6 +372,148 @@ static int guess_to_remote(const char *address, struct node_id *node_id,
 	return 1;
 }
 
+static void get_words(struct words **words) {
+	printf("Select your language:\n");
+	printf("  0) English (en)\n");
+	printf("  1) Spanish (es)\n");
+	printf("  2) French (fr)\n");
+	printf("  3) Italian (it)\n");
+	printf("  4) Japanese (jp)\n");
+	printf("  5) Chinese simplified (zhs)\n");
+	printf("  6) Chinese traditional (zht)\n");
+	printf("Select [0-6]: ");
+
+	char *selected = NULL;
+	size_t size = 0;
+	size_t characters = getline(&selected, &size, stdin);
+	if (characters < 0)
+		errx(ERROR_USAGE, "Could not read line from stdin.");
+
+	int selection = selected[0] - '0';
+	if (characters != 2 || selection < 0 || selection > 6)
+		errx(ERROR_USAGE, "You must select a number from the list.");
+
+	switch (selection) {
+		case 0:
+			bip39_get_wordlist("en", words);
+			break;
+		case 1:
+			bip39_get_wordlist("es", words);
+			break;
+		case 2:
+			bip39_get_wordlist("fr", words);
+			break;
+		case 3:
+			bip39_get_wordlist("it", words);
+			break;
+		case 4:
+			errx(ERROR_LANG_NOT_SUPPORTED, "Sorry, the language selected is not supported yet.");
+			bip39_get_wordlist("jp", words);
+			break;
+		case 5:
+			bip39_get_wordlist("zhs", words);
+			break;
+		case 6:
+			bip39_get_wordlist("zht", words);
+			break;
+		default:
+			errx(ERROR_USAGE, "Invalid language selection, select one from the list [0-6].");
+	}
+}
+
+static void get_mnemonic(char *mnemonic) {
+	char *line = NULL;
+	size_t line_size = 0;
+
+	printf("Introduce your BIP39 word list:\n");
```

Comment:
I'd add `separated by space`, so people don't attempt to write one per line, which is what I'd have done :-)